### ISSUE TITLE
Invalid XML declarations allowed

### ISSUE THREAD

**Issue by wt** [tafia/quick-xml#641]
The parser allows XML declarations anywhere in the file. If a decoration exists it should be at the very beginning of the file.

Also, only one declaration should be allowed.

Both of these constraints are not currently enforced.

**Comment by Mingun** [comment]
I think, this is by desing, but we can have a setting to enforce checks, and definitely should have this checks for serde deserializer

**Comment by wt** [comment]
It would also be useful to have a Reader/NSReader capable of enforcing such a thing for when I know there is only one document in a file/str source.

**Comment by Mingun** [comment]
Yeah, I say exactly this -- add a setting to enforce this behavior

**Comment by PhilipDaniels** [comment]
In addition to these problems, quick_xml also allows spurious additional attributes on the Xml declaration. The following is parsed fine into a `BytesDecl`:
```xml
    <?xml version="1.0" foo="bar"?>
```

According to the spec at https://www.w3.org/TR/REC-xml/#NT-prolog this should be forbidden.

**Comment by Mingun** [comment]
This is planned to be fixed in validation API the preliminary version of which can be found in my [validation](https://github.com/Mingun/quick-xml/tree/validation) branch
